Various/cool_MK ReSampler.lua

```diff
@@ -0,0 +1,5156 @@
+RandomPitch_Sld_state = tonumber(r.GetExtState('MK_ReSampler','RandomPitch_Sld.norm_val'))or 0;  
+BaseOctave_state = tonumber(r.GetExtState('MK_ReSampler','BaseOctave.norm_val'))or 4;
+SingleVoice_state = tonumber(r.GetExtState('MK_ReSampler','SingleVoice.norm_val'))or 2;
+Speed_state = tonumber(r.GetExtState('MK_ReSampler','Speed.norm_val'))or 4;
+Notes_On = tonumber(r.GetExtState('MK_ReSampler','Notes_On'))or 1;
+
+if RememberLast == nil then RememberLast = 1 end 
+if RememberLast <= 0 then RememberLast = 0 elseif RememberLast >= 1 then RememberLast = 1 end  
+if WFiltering == nil then WFiltering = 1 end 
+if WFiltering <= 0 then WFiltering = 0 elseif WFiltering >= 1 then WFiltering = 1 end 
+
+
+------------------------------------GetItemByType-----------------------------------------------------
+function GetItemByType() -- Selected or on Specific Named Track
+local Table = {}
+local tracks, trk, retval, TName, items, item
+
+item = r.GetSelectedMediaItem(0,0) --
+    if item then
+      Table = {item = item}
+         else
+            tracks = r.CountTracks();
+            for i = tracks, 0, -1 do
+               trk = r.GetTrack(0, i)
+                  if trk then
+                     retval, TName = r.GetSetMediaTrackInfo_String(trk, "P_NAME", "", false) -- 
+                     if TName == track_name then 
+                            items   =   r.CountTrackMediaItems(trk)
+                            for i = 0, items do
+                                item = r.GetTrackMediaItem(trk, i)
+                                 if item then
+                                 Table = {item = item}
+                                 end
+                           end
+                     end
+                end
+           end
+   end
+
+return Table.item
+
+end
+
+
+---------------------------------------Check and Store Inits----------------------------------------------------------
+
+ loopcheck = 0
+----loopcheck------
+local loopcheckstart, loopcheckending = r.GetSet_LoopTimeRange( 0, true, 0, 0, 0 )
+if loopcheckstart == loopcheckending and loopcheckstart and loopcheckending then 
+     loopcheck = 0
+       else
+     loopcheck = 1
+end
+
+function GetLoopTimeRangeInit()
+   local st, en
+   local selection = {st, en}
+   st, en = r.GetSet_LoopTimeRange( 0, 0, 0, 0, 0 )
+   selection[1] = {st = st}
+   selection[2] = {en = en}
+   start_init = selection[1].st
+   ending_init = selection[2].en
+end
+GetLoopTimeRangeInit()
+
+if FontAntiAliasing == 1 then
+    GFX2IMGUI_NO_LOG = true
+    GFX2IMGUI_MAX_DRAW_CALLS = 1<<16
+    local gfx2imgui_path = r.GetResourcePath() .. '/Scripts/ReaTeam Extensions/API/gfx2imgui.lua'
+    local os_sep = package.config:sub(1,1)
+    gfx2imgui_path = gfx2imgui_path:gsub( "/", os_sep )
+    if r.file_exists( gfx2imgui_path ) then
+       gfx = dofile(r.GetResourcePath() .. '/Scripts/ReaTeam Extensions/API/gfx2imgui.lua')
+       RG_status = "(RealmGUI)"
```

Comment:
ReaImGui: rea im gui (stands for "reaper immediate-mode graphical user interface"), not "realm gui". 😉